sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/internal/ServiceInstanceIdResourceProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure.internal;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ConditionalResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.UUID;
+
+/**
+ * does not implement {@link ResourceProvider}, because it depends on all attributes discovered by
+ * the other providers.
+ */
+public final class ServiceInstanceIdResourceProvider implements ConditionalResourceProvider {
+
+  public static final AttributeKey<String> SERVICE_INSTANCE_ID =
+      AttributeKey.stringKey("service.instance.id");
+
+  // multiple calls to this resource provider should return the same value
+  public static final Resource RANDOM =
```

Comment:
No need to make this public - we generally try to avoid public constants. Someone can call `new ServiceInstanceIdResourceProvider().createResource()` to access. If we receive user feedback that this is not enough, we should add `public static Resource get()` (or similarly named). But let's wait for feedback.